src/types/lib/contract/abi.ts

```diff
@@ -48,7 +48,9 @@ export type EnumAbi = {
 
 export type AbiEvents = { [hash: string]: EventAbi };
 
-export type EventAbi = Cairo1Event | LegacyEvent;
+export type EventAbi = Cairo1Events | LegacyEvent;
+
+export type Cairo1Events = Cairo1Event | AbiEvents;
```

Comment:
PR is LGTM, just some polishing.

- Rename all Cairo1 to Cairo. (we can consider old Cairo Cairo0 or Legacy and all new one's just Cairo)

Reply:
One more thing:
```ts
export type AbiEvents = { [hash: string]: EventAbi }; // Expect from name this are definitions of events in ABI
export type EventAbi = Cairo1Events | LegacyEvent; // Except from name this is definition of one event in abi with swaped words, but it is multiple onchain events for Cairo1 and singular onchain event for Cairo0
export type Cairo1Events = Cairo1Event | AbiEvents; // Then here we have onchain actual events that are > Cairo1 singular event or multiple ABI definitions of events ?


export type AbiStructs = { [name: string]: StructAbi }; // This name swaping for singular is confusing as hell 
export type AbiEnums = { [name: string]: EnumAbi }; // This name swaping for singular is confusing as hell

```

Can we resolve this namings its super confusingn hard to figure out what is what.
We already have a breaking changes so let batch tham

Reply:
One more thing
```ts
export type AbiEvents = { [hash: string]: EventAbi }; // Expect from name this are definitions of events in ABI, This name swaping for singular is confusing as hell 
export type EventAbi = Cairo1Events | LegacyEvent; // Except from name this is definition of one event in abi with swaped words, but it is multiple onchain events for Cairo1 and singular onchain event for Cairo0
export type Cairo1Events = Cairo1Event | AbiEvents; // Then here we have onchain actual events that are > Cairo1 singular event or multiple ABI definitions of events ?
export type AbiStructs = { [name: string]: StructAbi }; // This name swaping for singular is confusing as hell 
export type AbiEnums = { [name: string]: EnumAbi }; // This name swaping for singular is confusing as hell
```

All this is super confusing let batch breaking changes and resolve this also.

Reply:
Also, this ABI Spec stuff should be used from API/spec instead of redefining locally

Reply:
AbiStructs & AbiEnums are coded this way for a long time.
I can change all this, but I am not sure at all to have understood exactly what you are expecting.
I will try something, but you will have probably to finalize, because these things are so personal, and I have not access to your brain (probably a good thing for everybody...).

Reply:
```
export type Cairo1Events = Cairo1Event | AbiEvents
```

A Cairo 1  Abi includes 2 types of events definition :
- The detailed definition of each event. includes`members` key.
- The list of the events included in a component or in the main code. Descriptions can be nested. includes `variants` key.